Basic Unit Test for Verhoeff

tersal opened the pull request:
Solves #37114 

Basically is the creation of a new set of unit tests focused on the Verhoeff algorithm for checksums.

#### Testing

Tested with the script build_coverage.sh as part of the Unit Tests.

tersal committed:
Create basic Unit Test for Verhoff

tersal committed:
Change the build script for Accessor class

tersal committed:
Change structure and remove private accessors

tersal committed:
Change dependencies in build file

andy31415 commented on src/lib/support/verhoeff/tests/BUILD.gn:
```diff
@@ -0,0 +1,39 @@
+  sources = []
+
+  public_deps = [
+    "${chip_root}/src/app/common:cluster-objects",
```
this dependency list feels a bit big. Why do we need cluster objects for example? can we clean  it up?

andy31415 replied:
I believe you only need to depend on support is sufficient. I believe right now you get `support:suppport` by transitive deps as a sideffect.

tersal committed:
Reduce dependencies list

andy31415 committed:
Remove empty sources

Editorial commit: remove extra `sources = []`, I expect our templates to handle this one missing.
